### ISSUE TITLE
Add support for rust version for remap-debug-info=true option

### ISSUE THREAD

**Issue by psandana** [taiki-e/cargo-llvm-cov#311]
On this section ([link](https://github.com/taiki-e/cargo-llvm-cov/blob/3b9d2ca321a37cc35d42f9bc3e4fdca9999c58c3/src/main.rs#L1083C25-L1083C25)):

```rust
 if !cx.args.cov.disable_default_ignore_filename_regex {
        // TODO: Should we use the actual target path instead of using `tests|examples|benches`?
        //       We may have a directory like tests/support, so maybe we need both?
        if cx.args.remap_path_prefix {
            out.push(format!(
                r"(^|{SEPARATOR})(rustc{SEPARATOR}[0-9a-f]+|tests|examples|benches){SEPARATOR}"
            ));
        } else {
            out.push(format!(
                r"{SEPARATOR}rustc{SEPARATOR}[0-9a-f]+{SEPARATOR}|^{}({SEPARATOR}.*)?{SEPARATOR}(tests|examples|benches){SEPARATOR}",
                regex::escape(cx.ws.metadata.workspace_root.as_str())
            ));
        }
```
The remap-debug-info=true option is handled only for sha-only versions of rustc. But, while building rust, if rust fails to detect a commit-ish sha to sue, it will default to use the rustc version as shown [here](https://github.com/rust-lang/rust/blob/26089ba0a2d9dab8381ccb0d7b99e704bc5cb3ed/src/bootstrap/lib.rs#L1148C18-L1148C18).

```rust
fn debuginfo_map_to(&self, which: GitRepo) -> Option<String> {
        if !self.config.rust_remap_debuginfo {
            return None;
        }

        match which {
            GitRepo::Rustc => {
                let sha = self.rust_sha().unwrap_or(&self.version);
                Some(format!("/rustc/{sha}"))
            }
            GitRepo::Llvm => Some(String::from("/rustc/llvm")),
        }
    }
```

It is recommended to add version support along the sha commit, for rust builds which cannot obtain the rust_sha() value.

**Comment by taiki-e** [comment]
Thanks for the investigation! It definitely makes sense for us to support it since it comes from the way rustc works.

**Comment by taiki-e** [comment]
Fixed in 0.5.32.
